src/math/log_10.c

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: BSD-3-Clause/
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+#include <sof/math/log.h>
+#include <sof/audio/format.h>
+/**
+ * Base-10 logarithm  log10(x)
+ *
+ * loge = (u) computes the base-10 logarithm of
+ * u using lookup table.
+ * input u must be scalar/real number and positive
+ *
+ * +------------------+-----------------+--------+--------+
+ * | inpfxp	      |log10(returntype)| inpfxp |  loge  |
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * | 32 | 0   |	 0    | 32 | 28 | 0     | 32.0	 | 4.28   |
+ * +------------------+-----------------+--------+--------+
+ * Arguments	: uint32_t numerator [1 to 4294967295, Q32.0]
+ * Return Type	: uint32_t UQ4.28 [0 to 9.6329499409]
+ */
+uint32_t log10_int32(uint32_t numerator)
```

Comment:
You still need to add the `inline` keyword to get the optimization

Reply:
I compared gcc and clang outputs a number of times and I've never seen `inline` making any difference, at least never with `-On` where n > 0. The optimizer always thinks it knows better. That's why there is non-portable stuff like ALWAYS_INLINE.

(`static inline` is still useful in .h files to silence some warnings)

Reply:
given we know xcc isnt the smartest sometimes i think it is still wise to have the hints for it

Reply:
if I change `inline ln_int32()` and `inline log10_int32()`, I could see below error
```
No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory
ERROR: inline keyword should sit between storage class and type
#10: FILE: src/math/log_10.c:28:
+uint32_t inline log10_int32(uint32_t numerator)

ERROR: inline keyword should sit between storage class and type
#23: FILE: src/math/log_e.c:29:
+uint32_t inline ln_int32(uint32_t numerator)
```
In case I make changes in `log.h`  like below 
```
static inline uint32_t ln_int32(uint32_t numerator)
{
	return((uint32_t)Q_SHIFT_RND((int64_t)base2_logarithm(numerator) *
				     ONE_OVER_LOG2_E, 64, 32));
}
static inline uint32_t log10_int32(uint32_t numerator)
{
		return((uint32_t)Q_SHIFT_RND((int64_t)base2_logarithm(numerator) *
				     ONE_OVER_LOG2_10, 63, 32));
}
```
then `log10_int32(uint32_t numerator)` and `ln_int32(uint32_t numerator)` in `log_10.c` and `log_e.c` will be redundant , the impact of removing  `log_10.c` and `log_e.c` will result in error in cmocka compilation. cmocka `CmakeList.txt` is making use of `log_10.c` and `log_e.c` files.

```
cmocka_test(base_10_logarithm
	base_10_logarithm.c
	${PROJECT_SOURCE_DIR}/src/math/log_10.c
	${PROJECT_SOURCE_DIR}/src/math/base2log.c
)

cmocka_test(base_e_logarithm
	base_e_logarithm.c
	${PROJECT_SOURCE_DIR}/src/math/log_e.c
	${PROJECT_SOURCE_DIR}/src/math/base2log.c
)

```